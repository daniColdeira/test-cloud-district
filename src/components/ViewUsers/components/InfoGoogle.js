import React, {useContext} from 'react';
import { store } from '../../../store/store.js';
import Logout from '../../GoogleLogout';
import { InfoGoogle, DivGoogle } from '../styled';

function Info() {
    const globalState = useContext(store);
    
    return (
        <InfoGoogle>
            <img src={globalState.state.user.profileObj && globalState.state.user.profileObj.imageUrl} alt="logo" />
            <DivGoogle>
                <div>
                    {globalState.state.user.profileObj && globalState.state.user.profileObj.name}
                </div>
                <div>
                    {globalState.state.user.profileObj && globalState.state.user.profileObj.email}
                </div>
            </DivGoogle>
            <Logout />
        </InfoGoogle>
    );
}

export default Info;