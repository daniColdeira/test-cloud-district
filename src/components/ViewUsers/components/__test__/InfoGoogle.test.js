import React from 'react';
import { render } from '@testing-library/react';
import InfoGoogle from '../InfoGoogle';
import renderer from 'react-test-renderer';
import Provider, {AuthContext} from '../../../../AuthContext/AuthContext.js';
import TestRenderer from 'react-test-renderer';
import { Router } from "react-router-dom"; //enrutamiento
import historyRouting from "../../../../history"; //historial de rutas

test("renders correctly", () => {
    const tree = renderer
    .create(
        <Provider>
            <Router history={historyRouting}>
                <InfoGoogle />
            </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});



test("render with user", () => {
    let setUserAuth = jest.fn();
    let user = { user: { profileObj:{ name:'Daniel Coldeira', email:'danielcoldeirarm@gmail.com'}}}
    const element = new TestRenderer.create(
        <AuthContext.Provider value={{ user, setUserAuth }}>
            <InfoGoogle />
        </AuthContext.Provider>
    );
    expect(element).toMatchSnapshot();
});