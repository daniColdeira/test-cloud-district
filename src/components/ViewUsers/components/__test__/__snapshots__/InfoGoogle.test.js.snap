// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with user 1`] = `
<div
  className="sc-hBMVcZ igCLLv"
>
  <img
    alt="logo"
  />
  <div
    className="sc-kfYqjs fDwxtA"
  >
    <div>
      Daniel Coldeira
    </div>
    <div>
      danielcoldeirarm@gmail.com
    </div>
  </div>
  <button
    className="sc-eCApGN kBtpMM"
    onClick={[Function]}
  >
    <span>
      LogOut
    </span>
  </button>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="sc-hBMVcZ igCLLv"
>
  <img
    alt="logo"
  />
  <div
    className="sc-kfYqjs fDwxtA"
  >
    <div />
    <div />
  </div>
  <button
    className="sc-eCApGN kBtpMM"
    onClick={[Function]}
  >
    <span>
      LogOut
    </span>
  </button>
</div>
`;
